test/integration/options/posixExitCodes.spec.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+var helpers = require('../helpers');
+var runMocha = helpers.runMocha;
+
+describe('--posix-exit-codes', function () {
+  describe('when enabled, and with node options', function () {
+    var args = ['--no-warnings', '--posix-exit-codes'];
+
+    it('should exit with code 134 on SIGABRT', function (done) {
+      var fixture = 'signals-sigabrt.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 134);
+        done();
+      });
+    });
+
+    it('should exit with code 143 on SIGTERM', function (done) {
+      var fixture = 'signals-sigterm.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 143);
+        done();
+      });
+    });
+
+    it('should exit with code 0 even if there are test failures', function (done) {
+      var fixture = 'failing.fixture.js';
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', 0);
+        done();
+      });
+    });
+  });
+
+  describe('when not enabled, and with node options', function () {
+    it('should exit with code null on SIGABRT', function (done) {
+      var fixture = 'signals-sigabrt.fixture.js';
+      var args = ['--no-warnings'];
+      runMocha(fixture, args, function postmortem(err, res) {
+        if (err) {
+          return done(err);
+        }
+        expect(res.code, 'to be', null);
+        done();
+      });
+    });
+
+    it('should exit with the number of failed tests', function (done) {
```

Comment:
it('should exit abnormally with EXIT_FAILURE when any tests fail (or if none performed)',

Reply:
When we don't specify `--posix-exit-codes` (which might better be called `--unix-exit-codes` based on the discussion here)... we want to leave the legacy behavior unchanged. The reason why we `expect(res.code).to.be(1)` is because there's exactly one failing test in that suite.  I should probably change it to be several failing tests and a few passing ones just to avoid confusion, because the `1` here isn't meant to be synonymous with the EXIT_FAILURE (ie. 1) general error code.

Reply:
having multiple failures would have made this more apparent.

Reply:
Yep, added more failures in the test fixture.